src/Avalonia.Base/Threading/Dispatcher.Invoke.cs

```diff
@@ -627,14 +627,20 @@ public void Post(SendOrPostCallback action, object? arg, DispatcherPriority prio
     }
 
     /// <summary>
-    /// Returns a task awaitable that would invoke continuation on specified dispatcher priority
+    /// Returns a task that would invoke continuation on specified dispatcher priority
     /// </summary>
-    public DispatcherPriorityAwaitable AwaitWithPriority(Task task, DispatcherPriority priority) =>
-        new(this, task, priority);
+    public async Task AwaitWithPriority(Task task, DispatcherPriority priority)
+    {
+        await Resume(priority);
+        await task;
+    }
     
     /// <summary>
-    /// Returns a task awaitable that would invoke continuation on specified dispatcher priority
+    /// Returns a task that would invoke continuation on specified dispatcher priority
     /// </summary>
-    public DispatcherPriorityAwaitable<T> AwaitWithPriority<T>(Task<T> task, DispatcherPriority priority) =>
-        new(this, task, priority);
+    public async Task<T> AwaitWithPriority<T>(Task<T> task, DispatcherPriority priority)
+    {
+        await Resume(priority);
```

Comment:
Subtle semantics change (2 operations get queued on dispatcher)

Reply:
It can be rewritten as:

```csharp
var result = await task.ConfigureAwait(false);

await Resume(priority);
return result;
```

It gives only one run on the scheduler.

Reply:
And if the current logic is awaiting only a non-completed task with the priority, but otherwise immediately return a value, then:

```csharp
var completed = task.IsCompleted;
var result = await task.ConfigureAwait(false);

if (!completed)
    await Resume(priority);

return result;
```